api/app/zonal_stats.py

```diff
@@ -250,18 +253,33 @@ def calculate_stats(
     if group_by:
         zones_filepath = _group_zones(zones_filepath, group_by)
 
-    stats_input: FilePath | list = zones_filepath
+    zones_geojson: GeoJSON = get_json_file(zones_filepath)
+
+    if filter_by is not None:
+        key, value = filter_by
+        stats_input: list[Geometry] = [
+            f["geometry"]
+            for f in zones_geojson["features"]
+            if key in f["properties"].keys() and str(f["properties"][key]) == value
+        ]
+
+        if len(stats_input) == 0:
+            message = f"Property value '{value}' not found"
+            logger.info(message)
+            raise HTTPException(status_code=404, detail=message)
```

Comment:
Make sure to use the forwarding mechanism implemented in https://github.com/WFP-VAM/prism-app/pull/605

Reply:
The forwarding mechanims checks for http statuses. Not sure how to implement it here since, we are not making requests to servers, rather than filtering based on feature properties.

Reply:
Yes you're right, my bad :)